IOS/SwiftSeedProject/SwiftSeedProject/Infrastructure/Network/Api.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  Api.swift
+//  SwiftSeedProject
+//
+//  Created by Brian Sztamfater on 4/4/17.
+//  Copyright © 2017 Making Sense. All rights reserved.
+//
+
+import Foundation
+import Alamofire
+import SwiftyJSON
+
+final class Api {
+    // MARK: - Private Properties
+    private let manager: ApiManagerProtocol
+    
+    // MARK: - Designated Initializer
+    init(manager: ApiManagerProtocol = Configuration.manager) {
+        self.manager = manager
+    }
+    
+    // MARK: - Public Methods
+    func getUser(userId: String, completion: @escaping (ApiResult<User>) -> Void) {
+        let _ = manager.apiRequest(endpoint: .GetUser(userId: userId)).apiResponse { response in
+            switch response.result {
+                case .success(let json):
+                    let user = User(json: json)
+                    completion(ApiResult{ return user })
+                case .failure(let error):
+                    completion(ApiResult{ throw error })
+            }
+        }
+    }
+    
+    func updateUser(user: User, completion: @escaping (ApiResult<User>) -> Void) {
+        let _ = manager.apiRequest(endpoint: .UpdateUser(userId: user.id!), parameters: user.toParameters()).apiResponse { response in
+            switch response.result {
+                case .success(let json):
```

Comment:
this code is identical to the getUser and createUser method. It will not be possible to centralize the code within the same method and depending on the result of each call, call the necessary method?